### PR TITLE
Bind the tag length into the CTR+HMAC AEAD computation

### LINKED ISSUE #108: Bind tag size into CTR + HMAC authentication
Right now, the ciphertexts produced by the CTR+HMAC ciphers can be converted into one another, at least in the longer-to-shorter direction.  This means that an attacker can downgrade security to the lowest level accepted by a receiver.  There are two straightforward ways to fix this:

```
 # Incorporate in the key derivation
-aead_secret = HKDF-Extract(sframe_key, 'SFrame10 AES CTR AEAD')
+aead_secret = HKDF-Extract(sframe_key, 'SFrame10 AES CTR AEAD ' + encode_big_endian(tag_len))

 # Incorporate in the MAC input
-auth_data = aad_len + ct_len + nonce + aad + ct
+auth_data = aad_len + ct_len + tag_len + nonce + aad + ct
```

I would lean toward incorporating it at the key derivation stage, since this would make the whole ciphertext different for two different tag lengths, not just the tag.

### PR DESCRIPTION
**Pull request by bifurcation**
Fixes #108, using both of the approaches described (belt and suspenders!)

### CODE REVIEW

File: draft-ietf-sframe-enc.md [hunk 1]
```diff
@@ -548,7 +548,10 @@ follows, where `Nk` represents the key size for the AES block cipher in use and
 
 ~~~~~
 def derive_subkeys(sframe_key):
-  aead_secret = HKDF-Extract(sframe_key, 'SFrame10 AES CTR AEAD')
+  # tag_len = length of a tag for this cipher, in bytes
```
Comment (GhostofCookie) [inline draft-ietf-sframe-enc.md:551]: Where is it that this is intended to be defined? Perhaps it doesn't matter in this example?

File: draft-ietf-sframe-enc.md [hunk 2]
```diff
@@ -562,7 +565,9 @@ to a number of bytes `tag_len` fixed by the ciphersuite.
 def compute_tag(auth_key, nonce, aad, ct):
   aad_len = encode_big_endian(len(aad), 8)
   ct_len = encode_big_endian(len(ct), 8)
-  auth_data = aad_len + ct_len + nonce + aad + ct
+  # tag_len = length of a tag for this cipher, in bytes
```
Comment (GhostofCookie) [inline draft-ietf-sframe-enc.md:568]: Same question above.

File: draft-ietf-sframe-enc.md (hunk 1 above)
Comment (bifurcation) [reply draft-ietf-sframe-enc.md]: It's the value `Nt` in the IANA table.

File: draft-ietf-sframe-enc.md (hunk 2 above)
Comment (bifurcation) [reply draft-ietf-sframe-enc.md]: Same answer :)

**Pull request merged**